### PR TITLE
buildkite serverless test

### PR DESCRIPTION
**Pull request by kaisecheng**
This commit adds a new Buildkite pipeline to test against serverless endpoint daily
There are two types of tests, rspec and bash scripts.
rspec runs parts of the existing integration test in logstash-core
bash scripts run Logstash with predefined pipelines and 
check result against `_node/stats` API

Tests cover 
- es-output
- es-input
- es-filter
- central pipeline management
- legacy monitoring
- dlq
- integration-filter
- kibana api
- metricbeat stack monitoring

### CODE REVIEW

**Comment by kaisecheng** [comment]
sample buildkite [build](https://buildkite.com/elastic/logstash-serverless-integration-testing/builds/53#01894574-2906-4af4-8c44-acecdf95cf10)

File: .buildkite/serverless_integration_pipeline.yml [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/es_input_tests.sh
+  - label: "es-filter test"
+    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/es_filter_tests.sh
+  - label: "elastic_integration filter test"
+    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/elastic_integration_filter_tests.sh
+  - label: "central pipeline management test"
+    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/cpm_tests.sh
+  - label: "Logstash legacy monitoring test"
+    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/monitoring_tests.sh
+  - label: "kibana api test"
```
Comment (jsvd) [inline .buildkite/serverless_integration_pipeline.yml:21]:
```suggestion
  - label: "Kibana API test"
```

File: ci/serverless/common.sh [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  export ES_PW=$(vault read -field=es_user_pw "${vault_path}")
+  export KB_ENDPOINT=$(vault read -field=kb_host "${vault_path}")
+  set -x
+}
+
+build_logstash() {
+  ./gradlew clean bootstrap assemble installDefaultGems
+}
+
+prepare_test_data() {
```
Comment (jsvd) [inline ci/serverless/common.sh:25]: let's use a verb that better indicates the kind of preparation we're doing (indexing data to ES):

```suggestion
index_test_data() {
```

File: ci/serverless/cpm_tests.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+index_pipeline() {
+  RESP_CODE=$(curl -s -w "%{http_code}" -X PUT -u "$ES_USER:$ES_PW" "$ES_ENDPOINT/_logstash/pipeline/$1"  -H 'Content-Type: application/json' -d "$2")
+  if [[ $RESP_CODE -ge '400' ]]; then
+    echo "failed to update pipeline for Central Pipeline Management. Got $RESP_CODE from Elasticsearch"
+    exit 1
+  fi
+}
+
+# index pipeline to serverless ES
+prepare_cpm_pipelines() {
```
Comment (jsvd) [inline ci/serverless/cpm_tests.sh:18]:
```suggestion
index_cpm_pipelines() {
```

File: ci/serverless/dlq_rspec_tests.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+set +x
+export ES_ENDPOINT=$(vault read -field=es_host "${vault_path}")
+export ES_USER=$(vault read -field=es_user "${vault_path}")
+export ES_PW=$(vault read -field=es_user_pw "${vault_path}")
+set -x
+
+./gradlew clean bootstrap assemble installDefaultGems unpackTarDistribution
+./gradlew :logstash-core:copyGemjar
+
+export GEM_PATH=vendor/bundle/jruby/2.6.0
```
Comment (jsvd) [inline ci/serverless/dlq_rspec_tests.sh:19]: given that main is now on jruby 9.4 that uses a ruby 3.1 compatible implementation, this should no longer be working. does that mean that we don't need these two lines?

File: ci/serverless/elastic_integration_filter_tests.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+set -ex
+
+source ./$(dirname "$0")/common.sh
+
+prepare_ingest_pipeline() {
```
Comment (jsvd) [inline ci/serverless/elastic_integration_filter_tests.sh:6]:
```suggestion
deploy_ingest_pipeline() {
```

File: ci/serverless/pipeline/001_es-output.conf [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+output {
+    elasticsearch {
+        id => "named_index"
+        hosts => ["${ES_ENDPOINT}"]
+        user => "${ES_USER}"
+        password => "${ES_PW}"
+        index => "${INDEX_NAME}"
+
+        sniffing => true
```
Comment (jsvd) [inline ci/serverless/pipeline/001_es-output.conf:20]: No need to test sniffing in a cloud deployment (stateful or stateless)

```suggestion
```

File: ci/serverless/test_data/book.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{"index": {}}
```
Comment (jsvd) [inline ci/serverless/test_data/book.json:1]: damn you really went for the classics

Comment (kaisecheng) [reply ci/serverless/test_data/book.json:1]: just random data from www :)

File: ci/serverless/dlq_rspec_tests.sh (hunk 4 above)
Comment (kaisecheng) [reply ci/serverless/dlq_rspec_tests.sh]: tried to remove the lines but it fails to resolve rspec. Strange that it is required. I updated it to 3.1.0

File: ci/serverless/dlq_rspec_tests.sh [hunk 8]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -ex
+
+vault_path=secret/ci/elastic-logstash/serverless-test
+# Since we are using the system jruby, we need to make sure our jvm process
```
Comment (jsvd) [inline ci/serverless/dlq_rspec_tests.sh:5]: It's not system ruby, right? we're using the one we put under `vendor/` during the gradle tasks

**Review by jsvd — APPROVED** [review]
LGTM

**Pull request merged**